src/distribution/chi_squared.rs

```diff
@@ -310,7 +310,11 @@ impl Continuous<f64, f64> for ChiSquared {
     ///
     /// where `k` is the degrees of freedom and `Γ` is the gamma function
     fn pdf(&self, x: f64) -> f64 {
-        self.g.pdf(x)
+        if x > 0.0 {
```

Comment:
I think we can be more specific, `x == 0.0` is defined (at least according to Wikipedia) as long as `freedom != 1.0`, so we can probably do something like `if x > 0.0 || (self.freedom != 1.0 && x == 0.0)`. May want to consider `prec::almost_eq` instead of `==` or `!=` for floats but I'm just spitballing

Reply:
Sorry, I screwed up here. Should've done better research.

The thing is a little bit more complicated. Wikipedia silently assumes only integer degrees of freedom. If we take `freedom` as a positive real, then we have 3 cases:
1. `0 < freedom < 2`: `pdf(0) = +∞`, hence `0` is excluded from the support
2. `freedom == 2`: `pdf(0) = 0.5`
3. `2 < freedom`: `pdf(0) = 0`

This is obvious from the pdf formula (see Wikipedia) and the `x^(k/2 - 1)` term. All other terms are non-zero for any `freedom`.

---

Anyway, I've tried out `scipy`'s implementation of chi-squared. Its behaviour is exactly as above (i.e. `+∞`, `0.5` and `0`).

---

So the question is, what to do for the case when `0 < freedom < 2`: Should we let the pdf function return `+∞` (as `scipy` does)? Or should we handle it as a separate case?

My personal preference right now is to return `+∞`, but document this behaviour in the documentation.

Reply:
Yeah that's fine by me as long as it's clear in the documentation

Reply:
Ok! Just to make sure: You prefer to return `+∞` when `0 < freedom < 2` and pdf is evaluated at `0`.

Then I'll rewrite the unit tests and update the docs.